app/src/main/java/org/gdg/frisbee/android/gde/GdeActivity.java

```diff
@@ -94,102 +84,66 @@ public void networkFailure(Throwable error) {
-            @Override
-            public void run() {
-                HashMap<String, GdeList> gdeMap = new HashMap<>();
-
-                for (Gde gde : directory) {
-                    if (gde.getProduct() == null) {
-                        continue;
-                    }
-                    for (String p : gde.getProduct()) {
-                        String product = p.trim();
-                        if (!gdeMap.containsKey(product)) {
-                            gdeMap.put(product, new GdeList());
-                        }
-
-                        gdeMap.get(product).add(gde);
-                    }
-                }
-
-                mViewPagerAdapter.addMap(gdeMap);
-                mViewPagerAdapter.notifyDataSetChanged();
-
-                mViewPager.setAdapter(mViewPagerAdapter);
-                mTabLayout.setupWithViewPager(mViewPager);
-
-                getAchievementActionHandler().handleLookingForExperts();
-            }
-        });
-    }
-
-    protected String getTrackedViewName() {
-        return null;
+    private void setupGdeViewPager(GdeList directory) {
+        // TODO use sorted HashMap to sort the categories.
+        HashMap<String, GdeList> gdeMap = extractCategoriesFromGdeList(directory);
+        List<GdeCategory> gdeCategoryList = convertCategoryMapToList(gdeMap);
+
+        mViewPagerAdapter = new GdeCategoryPagerAdapter(
+            getSupportFragmentManager(),
+            getString(R.string.about),
+            gdeCategoryList
+        );
+        mViewPager.setAdapter(mViewPagerAdapter);
+        mViewPager.addOnPageChangeListener(this);
+        mTabLayout.setupWithViewPager(mViewPager);
+
+        getAchievementActionHandler().handleLookingForExperts();
     }
 
-    public class GdeCategoryAdapter extends FragmentStatePagerAdapter implements ViewPager.OnPageChangeListener {
-        private final SparseArray<WeakReference<Fragment>> mFragments = new SparseArray<>();
-        private HashMap<String, GdeList> mGdeMap;
-
-        public GdeCategoryAdapter(FragmentManager fm) {
-            super(fm);
-            mGdeMap = new HashMap<>();
-        }
-
-        public void addMap(Map<String, GdeList> collection) {
-            mGdeMap.putAll(collection);
-        }
-
-        @Override
-        public int getCount() {
-            return mGdeMap.keySet().size() + 1;
-        }
-
-        @Override
-        public Fragment getItem(int position) {
-            if (position == 0) {
-                return PlainLayoutFragment.newInstance(R.layout.fragment_gde_about);
-            } else {
-                String key = mGdeMap.keySet().toArray(new String[mGdeMap.size()])[position - 1];
-                Fragment frag = GdeListFragment.newInstance(mGdeMap.get(key));
-                mFragments.append(position, new WeakReference<>(frag));
+    private HashMap<String, GdeList> extractCategoriesFromGdeList(GdeList directory) {
+        HashMap<String, GdeList> gdeMap = new HashMap<>();
```

Comment:
Unfortunately the diff here looks bad. 
This method is basically the same. I just extracted this into a method. 
This gets the whole list and split it into categories.